lib/custom_a11y_rules.js

```diff
@@ -0,0 +1,150 @@
+      "enabled": true,
+      "excludeHidden": false,
+      "tags": ["edx-custom"],
+      "all": [
+        "icon-aria-hidden"
+      ],
+      "any": [],
+      "none": [],
+      "metadata": {
+          "description": "Ensures all icon-font elements are aria-hidden",
+          "help": "All <i> elements should have aria-hidden=\"true\"",
+          "helpUrl": "https://openedx.atlassian.net/wiki/display/A11Y/edX+Specific+Accessibility+Tests"
+      }
+    }
+  ],
+  "checks": [
+    {
+      "id": "nav-aria-label-present",
+      "metadata": {
+        "impact": "critical",
+        "messages": {
+          "pass": "all <nav> elements MUST have an aria-label or aria-labelledby attribute",
+          "fail": "all <nav> elements MUST have aria-label or aria-labelledby attribute"
+        }
+      },
+      evaluate: function(node, options) {
+        return axe.commons.aria.label(node) !== null;
+      },
+    },
+    {
+      "id": "nav-aria-label-value",
+      "metadata": {
+        "impact": "critical",
+        "messages": {
+          "pass": "aria-label values on <nav> elements MUST NOT have \"navigation\" or \"nav\" as part of their value",
+          "fail": "aria-label values on <nav> elements MUST NOT have \"navigation\" or \"nav\" as part of their value"
+        }
+      },
+      evaluate: function(node, options) {
+        label = axe.commons.aria.label(node) || '';
+        words = label.toLowerCase().split(" ");
+        return words.indexOf('nav') < 0 && words.indexOf('navigation') < 0;
+      },
+    },
+    {
+      "id": "link-href",
+      "metadata": {
+        "impact": "critical",
+        "messages": {
+          "pass": "href value MUST NOT be empty or \"#\"",
+          "fail": "href value MUST NOT be empty or \"#\""
+        }
+      },
+      evaluate: function(node, options) {
+        href = node.getAttribute("href");
+        if (href === null || href === "" || href === "#") {
+          return false;
+        } else {
+          return true;
+        }
+      },
+    },
+    {
+      "id": "link-href-internal",
+      "metadata": {
+        "impact": "critical",
+        "messages": {
+          "pass": "internal link must have focusable target",
+          "fail": "internal link must have focusable target"
+        }
+      },
+      evaluate: function(node, options) {
+        href = node.getAttribute("href") || "";
+        if (href.startsWith("#") && href.length > 1) {
+          linkTarget = document.querySelector(href);
+          return axe.commons.dom.isFocusable(linkTarget);
+        } else {
+          return true;
+        }
+      },
+    },
+    {
+      "id": "icon-aria-hidden",
+      "metadata": {
+        "impact": "critical",
+        "messages": {
+          "pass": "icon elements should be aria-hidden",
+          "fail": "icon elements should be aria-hidden"
+        }
+      },
+      evaluate: function(node, options) {
+        return axe.utils.isHidden(node);
```

Comment:
I don't see the 'i' element being selected like how similar elements are selected in above tests...how is that done for this function?